### ISSUE TITLE
Why do you scale hyp['cls'] and hyp['obj']?

### ISSUE THREAD

**Issue by liudakai2** [ultralytics/yolov5#4904]
## ❔Question
It seems that in _train.py_ the hyper-parameters _hyp['cls']_ and _hyp['obj']_ are scaled by _class number_ and _image size_ independently. Now that the losses are finally averaged by tensor's size, is there any mathematical explanation for why these hyper-parameters should be scaled according to different datasets?

**Comment by glenn-jocher** [comment]
@liudakai2 yes, loss gains are adapted to training settings and dataset for improved performance.

**Comment by liudakai2** [comment]
Okay. Thank you.



---Original---
From: "Glenn ***@***.***&gt;
Date: Fri, Sep 24, 2021 23:37 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"Liu ***@***.***&gt;;
Subject: Re: [ultralytics/yolov5] Why do you scale hyp['cls'] and hyp['obj']? (#4904)




 
@liudakai2 yes, loss gains are adapted to training settings and dataset for improved performance.
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
